src/components/Post/Post.js

```diff
@@ -1,115 +1,106 @@
+      const {commentsSectionExpanded, error, isCommentsLoading, comments, commentsLoaded} = this.state;
+      const {item} = this.props;
+      const {id, title, body} = item;
 
-    return (
-      <div className={`${CN} card `}>
-        <div className={`${CN}__id`}>id: {id}</div>
-        <div className="card-body">
-          <div className="card-title"> {title} </div>
-          <div className="card-text text">{body}</div>
-        </div>
+      return (
+          <div className={`${CN} card `}>
+              <div className={`${CN}__id`}>id: {id}</div>
+              <div className="card-body">
+                  <div className="card-title"> {title} </div>
+                  <div className="card-text text">{body}</div>
+              </div>
 
-        {/* todo создать div который будет выполнять функцию кнопки */}
-        {/* todo у него должен быть класс, состоящий из комбинации базового класса CN и строки '__link-btn' */}
-        {/* todo в событие  onClick положить метод  onShowComments, объявленный выше */}
-        {/* todo в children этого div должна быть положена строка "Show comments" если флажок commentsSectionExpanded в стейте = true*/}
-        {/* todo иначе - строка "Hide comments" */}
+              <div className={`${CN}__link-btn`} onClick={this.onShowComments}>
+                  {commentsSectionExpanded ? "Hide comments" : "Show comments"}
+              </div>
+              <div>
+                  {error && <div>{error}</div>}
+              </div>
 
-        {/* todo создать div который будет выводить ошибку из стейта, если она существует (т.е. не пустая строка) */}
+              {commentsSectionExpanded && isCommentsLoading &&
+              <div className={`${CN}__loading`}>
+                  <LoadingIndicator/>
+              </div>}
 
-        {
-          //todo если секция комментариев открыта, т.е. commentsSectionExpanded = true
-          //   и идет загрузка комментариев, т.е. isCommentsLoading = true
-          // показываем лоадинг индикатор (можно просто строку с надписью "Loading comments ..." в div)
-          // если хотите, чтоб выглядело как в видео, добавьте к div класс состоящий из базового класса CN и "__loading"
-        }
-        {
-          //todo если секция комментариев открыта, т.е. commentsSectionExpanded = true
-          //   но НЕ идет загрузка комментариев, т.е. isCommentsLoading = false
-          //   и запрос уже был выполнен т.е. commentsLoaded = true
-          //   и массив comments пустой
-          // показываем сообщение, что нет результатов (строку с надписью "No comments for this post yet" в div)
-          // если хотите, чтоб выглядело как в видео, добавьте к div класс состоящий из базового класса CN и "__no-results"
-        }
-        {
+              {commentsSectionExpanded && !isCommentsLoading && commentsLoaded && !comments.length &&
+              <div className={`${CN}__no-results`}>{"No comments for this post yet"}</div>}
```

Comment:
Brackets are not needed here inside tag cause you are putting simple string as a child
can be just
`<div className={`${CN}__no-results`}>"No comments for this post yet"</div>`